### PR TITLE
[UI] Fix formatMetric to convert a small metric value into an exponential notation

### LINKED ISSUE #2006: [BUG] MLFlow UI run page shows all metric values rounded to 3 decimal points
### System information
- **OS Platform and Distribution (RHL 7.6)**:
- **MLflow installed from (binary, pip)**:
- **MLflow version (1.3.0)**:
- **Python version 3.6.8**:

### Describe the problem
While "all runs" MLFlow UI page shows metric values as they are reported (without rounding), individual run page displays all metrics rounded to 3 decimal points

### Other info / logs
all runs:
all_runs
run:
run

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Resolves #2006 

Screen Shot 2019-10-30 at 21 33 08

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s) does this PR affect?

- [x] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
I need to augment the test for `formatMetric` in `Utils.test.js`.

https://github.com/mlflow/mlflow/blob/master/mlflow/server/js/src/utils/Utils.test.js#L5

**Comment by harupy** [comment]
Related issue #1901

**Comment by harupy** [comment]
@smurching

**Review by Zangr — COMMENTED** [review]
LGTM generally! Could you please provide some comment & examples for this function as well?

**Comment by Zangr** [comment]
@gioa, what do you think about displaying ultra small number(< 0.001) with exponential notation?

**Comment by gioa** [comment]
@harupy Thank you for fixing the #2006! Is it possible to include the small number in the HTML title attribute so users can hover and see the value directly? I found myself manually calculated the value, I might not be the typical user here though. 😳

**Comment by harupy** [comment]
@gioa Let me try

**Comment by harupy** [comment]
@gioa what do you think?
title

**Comment by gioa** [comment]
@harupy Great! LGTM. Thank you for working on this!

**Comment by harupy** [comment]
@gioz @Zangr 
#1901 can be fixed the way as this one. Should I fix it in this PR?

**Comment by Zangr** [comment]
@harupy sure, please feel free to add the fix in this PR.

**Comment by harupy** [comment]
Metric values on experiment view

### before
before

### after
after

**Comment by aarondav** [comment]
This LGTM. @Zangr ?

**Review by aarondav — APPROVED** [review]

File: mlflow/server/js/src/components/BaggedCell.js [hunk 1]
```diff
@@ -76,8 +77,9 @@ export default class BaggedCell extends PureComponent {
             <span
               className="metric-param-value run-table-container"
               style={styles.metricParamCellContent}
+              title={value}
             >
-              {value}
+              {isParam ? value : Utils.formatMetric(parseFloat(value))}
```
[the remark this answers is not in the document]
Comment (aarondav) [reply mlflow/server/js/src/components/BaggedCell.js]: This seems like a reasonable first step, not all parameter values even if numerical should probably be exponential (e.g. user id)

[the remark this answers is not in the document]
Comment (Zangr) [reply mlflow/server/js/src/components/BaggedCell.js]: Make sense. In that case, I'd recommend this to be written in a more metric-centric condition:
`{isMetric ? Utils.formatMetric(parseFloat(value)) : value}`

[the remark this answers is not in the document]
Comment (aarondav) [reply mlflow/server/js/src/components/BaggedCell.js]: Agree, let's do `isMetric`

**Comment by aarondav** [comment]
LGTM @Zangr ?

**Review by Zangr — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
@aarondav #1901 also should be closed.

**Comment by rafalbachorz** [comment]
Hello,
I am aware that this has been close, I am know that the fix has been released in v 1.5.0. However, I use the 1.11.0 and I still see just 3 decimal digits. Shall I do something in order to increase the number of digits? I will appreciate the comments.

Kind regards,
Rafał

tmp

**Comment by aarondav** [comment]
This PR only improved handling of values <0.001 -- it does make sense that we might ensure some minimum number of sig figs, though, and I can't remember if we did anything towards that. Could you create an issue for this?

**Pull request merged**